VM/rel2vm_params.py

```diff
@@ -0,0 +1,1085 @@
+        help="max PGV at velocity model perimiter (estimated, cm/s)",
+        type=float,
+        default=5.0,
+    )
+    arg("--hh", help="velocity model grid spacing (km)", type=float, default=0.4)
+    arg(
+        "--dt",
+        help="timestep to estimate simulation duration (s)",
+        type=float,
+        default=0.005,
+    )
+    arg(
+        "--space-land",
+        help="min space between VM edge and land (km)",
+        type=float,
+        default=5.0,
+    )
+    arg(
+        "--space-srf",
+        help="min space between VM edge and SRF (km)",
+        type=float,
+        default=15.0,
+    )
+    arg("--min-vs", help="for nzvm gen and flo (km/s)", type=float, default=0.5)
+    arg("-n", "--nproc", help="number of processes", type=int, default=1)
+
+    arg("--vm-version", help="velocity model version to generate", default="1.65")
+    arg(
+        "--vm-topo",
+        help="topo_type parameter for velocity model generation",
+        default="BULLDOZED",
+    )
+    arg("--selection", help="also generate NHM selection file", action="store_true")
+    arg(
+        "--no_optimise",
+        help="Don't try and optimise the vm if it is off shore. Removes dependency on having GMT coastline data",
+        action="store_true",
+    )
+    arg(
+        "--min-rjb",
+        help="Specify a minimum horizontal distance (in km) for the VM to span from the fault"
+        " - invalid VMs will still not be generated",
+        default=0,
+    )
+    arg(
+        "--ds-multiplier",
+        help="Sets the DS multiplier for setting the sim-duration. Validation runs default to 1.2. Cybershake runs"
+        "should manually set it to 0.75",
+        default=1.2,
+        type=float,
+    )
+    args = parser.parse_args()
+    args.vm_root_dir = os.path.abspath(args.vm_root_dir)
+
+    return args
+
+
+if __name__ == "__main__":
+    logger = qclogging.get_logger("rel2vm_params")
+    qclogging.add_general_file_handler(
+        logger, os.path.join(os.getcwd(), "rel2vm_params_log.txt")
+    )
+    args = load_args(logger=logger)
+
+    logger.debug("Loading Srf/*.csv")
+    msg_list = load_msgs(args, logger=logger)
+    if len(msg_list) == 0:
+        message = "Found nothing to do, exiting."
+        logger.log(qclogging.NOPRINTCRITICAL, message)
+        sys.exit(message)
+
+    # prepare to run
+    if not os.path.isdir(args.vm_root_dir):
+        logger.debug(
+            "VM root directory {} does not exist. Creating it now.".format(
+                args.vm_root_dir
+            )
+        )
+        os.makedirs(args.vm_root_dir)
+
+    # distribute work
+    logger.debug(
+        "Number of processes to use given as {}, number of tasks: {}.".format(
+            args.nproc, len(msg_list)
+        )
+    )
+    p = Pool(processes=args.nproc)
+    reports = p.starmap(main, msg_list)
+
+    # store summary
+    store_summary(
+        os.path.join(args.vm_root_dir, "rel2vm_params_info.csv"), reports, logger=logger
+    )
```

Comment:
Now, it is handling one event and all output files go to the specified out dir. This is "summary" file for the particular event(fault), not for multiple events, meaning your proposed "append" option is not necessary.

Reply:
Is there anything in the summary for one event that isn't already in the vm_params file? We can get rid of the summary if not

Reply:
good point.

Reply:
If you think a summary is necessary, then it should be writing to a common folder so you can look at the entire dataset (such as sim durations across all faults in a run).

Reply:
It sounds like something that the "orchestration" script (to be developed) wants to do.

Reply:
Yep - as long as there is everything from the summary in the params file. Orchestration script is currently "low-priority" because of efforts to automate this.